### PR TITLE
Update perf_analyzer SSL docs

### PR DESCRIPTION
**Pull request by matthewkotila**

### CODE REVIEW

File: docs/perf_analyzer.md [hunk 1]
```diff
@@ -453,6 +453,28 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
 protocol can be specificed with the -i option. If GRPC is selected the
 --streaming option can also be specified for GRPC streaming.
 
+### SSL/TLS Support
+
+perf_analyzer allows communication accross a secured channel using both gRPC and HTTP protocols. Just setting the perf_analyzer SSL options does not ensure the secure communication. For gRPC, the Triton Server supports running with SSL/TLS enabled. For HTTP, the Triton server should be running behind `https://` proxy such as nginx. perf_analyzer can then establish a secure channel to the proxy.
+
+When running perf_analyzer with gRPC protocol, use any of the below options:
```
Comment (matthewkotila) [inline docs/perf_analyzer.md:460]: also add a link to this:

gRPC documentation on how they can use SSL without error and no SSL without error

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: docs/perf_analyzer.md [hunk 2]
```diff
@@ -453,6 +453,28 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
...
+
+* `--ssl-grpc-root-certifications-file`
```
[the remark this answers is not in the document]
Comment (matthewkotila) [reply docs/perf_analyzer.md:462]: can this just be inferred by the existence of any of the other options?

File: docs/perf_analyzer.md (hunk 1 above)
Comment (matthewkotila) [reply docs/perf_analyzer.md]: added comment and your link 👍

File: docs/perf_analyzer.md [hunk 3]
```diff
@@ -453,6 +453,28 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
 protocol can be specificed with the -i option. If GRPC is selected the
 --streaming option can also be specified for GRPC streaming.
 
+### SSL/TLS Support
+
+perf_analyzer allows communication accross a secured channel using both gRPC and HTTP protocols. Just setting the perf_analyzer SSL options does not ensure the secure communication. For gRPC, the Triton Server supports running with SSL/TLS enabled. For HTTP, the Triton server should be running behind `https://` proxy such as nginx. perf_analyzer can then establish a secure channel to the proxy.
```
[the remark this answers is not in the document]
Comment (matthewkotila) [reply docs/perf_analyzer.md]: reworded to what you put 👍

File: docs/perf_analyzer.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tanmayv25) [reply docs/perf_analyzer.md:462]: > You can use GRPC SSL without client certs.  Using `--ssl-grpc-use-ssl`, without any other options achieves something like this: https://grpc.io/docs/guides/auth/#using-client-side-ssltls

These three options makes sense here.

For advanced use cases such as modifying the root CA or using client certs, the corresponding options can be set in the SslCredentialsOptions parameter passed to the factory method.

File: docs/perf_analyzer.md [hunk 4]
```diff
@@ -453,6 +453,36 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
...
+
+perf_analyzer can be used to benchmark Triton service behind SSL/TLS-enabled endpoints. These options can help in establishing secure connection with the endpoint and profile the server.
+
+For gRPC, see the following options:
+
+* `--ssl-grpc-root-certifications-file`
+* `--ssl-grpc-private-key-file`
+* `--ssl-grpc-certificate-chain-file`
+
+More details here: https://grpc.github.io/grpc/cpp/structgrpc_1_1_ssl_credentials_options.html
```
Comment (tanmayv25) [inline docs/perf_analyzer.md:466]: You should use the hyperlink here.

File: docs/perf_analyzer.md [hunk 5]
```diff
@@ -453,6 +453,36 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
...
+
+The [inference protocol gRPC SSL/TLS section](inference_protocols.md#ssltls) describes server-side options to configure SSL/TLS in Triton's gRPC endpoint.
+
+For HTTPS, the following options are exposed:
+
+* `--ssl-http-verify-peer`
```
Comment (tanmayv25) [inline docs/perf_analyzer.md:472]: I was thinking.. wouldn't it make more sense to rename all these options as `--ssl-https_*`?  It becomes more clear that these options will be used for https endpoint only.

File: docs/perf_analyzer.md [hunk 6]
```diff
@@ -453,6 +453,36 @@ By default perf_analyzer uses HTTP to communicate with Triton. The GRPC
...
+* `--ssl-http-client-certificate-file`
+* `--ssl-http-client-certificate-type`
+* `--ssl-http-private-key-file`
+* `--ssl-http-private-key-type`
+
+See `--help` for full documentation.
+
+Unlike gRPC, Triton's HTTP server endpoint can not be configured with SSL/TLS support.
+
+Note: Just providing these `--ssl-http-*` options to perf_analyzer does not ensure the SSL/TLS is used in communication. If SSL/TLS is not enabled on the service endpoint, these options have no effect. The intent of exposing these options to a user of perf_analyzer is to allow them to configure perf_analyzer to benchmark Triton service behind SSL/TLS-enabled endpoints. In other words, if Triton is running behind a HTTPS server proxy, then these options would allow perf_analyzer to profile Triton via exposed HTTPS proxy.
```
Comment (tanmayv25) [inline docs/perf_analyzer.md:484]: change `--ssl-http-*` to `--ssl-https-*` as part of above renaming.

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: docs/perf_analyzer.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthewkotila) [reply docs/perf_analyzer.md:463]: added `--ssl-grpc-use-ssl` option 👍

File: docs/perf_analyzer.md (hunk 4 above)
Comment (matthewkotila) [reply docs/perf_analyzer.md:467]: GitHub markdown auto-renders this as a hyperlink 👍

File: docs/perf_analyzer.md (hunk 5 above)
Comment (matthewkotila) [reply docs/perf_analyzer.md]: renamed to `--ssl-https-*` 👍

File: docs/perf_analyzer.md (hunk 6 above)
Comment (matthewkotila) [reply docs/perf_analyzer.md:485]: renamed to `--ssl-https-*` 👍

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**